From 42663b51a01d5fa11675752338bc9d67f232e96f Mon Sep 17 00:00:00 2001
From: Bogdan Pricope <bogdan.pricope@linaro.org>
Date: Fri, 27 Jan 2017 13:57:26 +0200
Subject: [PATCH] Temporary change value of maximum number of cores parameter

Signed-off-by: Bogdan Pricope <bogdan.pricope@linaro.org>
---
 include/api/ofp_config.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/api/ofp_config.h b/include/api/ofp_config.h
index 492269d..0e8d98b 100644
--- a/include/api/ofp_config.h
+++ b/include/api/ofp_config.h
@@ -17,7 +17,7 @@
 
 /**Maximum number of CPUs.
  * Used to define the size of internal structures. */
-#define OFP_MAX_NUM_CPU 64
+#define OFP_MAX_NUM_CPU 32
 
 /* Enable features */
 
-- 
2.9.3

